src/type/scalars.ts

```diff
@@ -113,7 +118,7 @@ export const GraphQLFloat = new GraphQLScalarType<number>({
       num = Number(coercedValue);
     }
 
-    if (typeof num !== 'number' || !Number.isFinite(num)) {
+    if (!isNumeric(num)) {
```

Comment:
If I follow this correctly, the result of this coercion could be of JS type `bigint` rather than of type JS `number`.

This would seem to be unexpected.

maybe the line above can be edited to:

>     if (
>      typeof coercedValue === 'bigint' ||
>      (typeof coercedValue === 'string' && coercedValue !== '')
>    ) {

Note: there is a loss of precision when convering from bigint => float. Another option is to error, so that developers can do the conversion on their own and be more aware of the loss of precision.